### PR TITLE
Feat/722: Network switcher in console2

### LINKED ISSUE #722: Switch to another network from Console
Console should have a way of getting from the network you are on to another network.

for example: you've gone to console.fairground.wtf but now you want to jump to mainnet. There should be some UI that allows you to jump to another network. This to be done by jumping to another URL (as apposed to connecting to a different data node).

Note: the [Node switcher](https://github.com/vegaprotocol/frontend-monorepo/issues/622) exists and should be thought about in context of this.

---

e.g. This is a very rough sketch and needs to be developed, and alternatives considered...

Image

### PR DESCRIPTION
**Pull request by notbot00**
# Related issues 🔗

Closes #722 

# Description ℹ️

Adds a network switcher to the trading app. Also added stagnet3 support to the environment lib.

# Demo 📺

Screenshot 2022-08-18 at 12 53 58

### CODE REVIEW

File: apps/trading/components/navbar/navbar.tsx [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-      <Link href="/" passHref={true}>
-        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
-        <a className="px-[26px]">
-          <Vega className="fill-white" />
-        </a>
-      </Link>
+      <div className="flex items-center h-full">
+        <Link href="/" passHref={true}>
+          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+          <a className="px-[26px]">
```
Comment (sam-keen) [inline apps/trading/components/navbar/navbar.tsx:17]: Suggestion (non-blocking): absolutely fine if 26px exactly is needed, but if not, you could use either 24px or 28px from the theme

File: libs/environment/src/components/network-switcher/network-switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+export const envTriggerMapping: Record<Networks, string> = {
+  ...envNameMapping,
+  [Networks.TESTNET]: t('Fairground'),
+};
+
+export const envDescriptionMapping: Record<Networks, string> = {
+  [Networks.CUSTOM]: '',
+  [Networks.DEVNET]: t('The latest Vega code auto-deployed'),
+  [Networks.STAGNET]: t('A staging environment with trading'),
+  [Networks.STAGNET2]: t('A staging envirnment without trading'),
```
Comment (sam-keen) [inline libs/environment/src/components/network-switcher/network-switcher.tsx:34]:
```suggestion
  [Networks.STAGNET2]: t('A staging environment without trading'),
```

File: libs/environment/src/components/network-switcher/network-switcher.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  isCurrent: boolean;
+  isAvailable: boolean;
+};
+
+const NetworkLabel = ({
+  isCurrent = false,
+  isAvailable = false,
+}: NetworkLabelProps) => (
+  <span className="text-white-80 dark:text-white-80">
+    {isCurrent
```
Comment (sam-keen) [inline libs/environment/src/components/network-switcher/network-switcher.tsx:64]: Suggestion (non-blocking): I think nested ternaries are a bit unpleasant to read. I have a small preference for this render logic to sit outside JSX and be done with if/else. Not a strong objection though :)

File: libs/environment/src/components/network-switcher/network-switcher.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    (isOpen: boolean) => {
+      setOpen(isOpen);
+      if (!isOpen) {
+        setAdvancedView(false);
+      }
+    },
+    [setOpen, setAdvancedView]
+  );
+
+  const menuItemClasses = 'pt-12 pb-12 pl-16 pr-16 h-auto';
```
Comment (sam-keen) [inline libs/environment/src/components/network-switcher/network-switcher.tsx:87]:
```suggestion
  const menuItemClasses = 'py-12 px-16 h-auto';
```

**Review by sam-keen — APPROVED** [review]
Couple of minor suggestions, looks good

**Review by mattrussell36 — APPROVED** [review]

File: libs/environment/src/components/network-switcher/network-switcher.tsx (hunk 4 above)
Comment (notbot00) [reply libs/environment/src/components/network-switcher/network-switcher.tsx:84]: this doesn't work, as the dropdown classes override them, that's why i had to hack it in this way. or i could also potentially just not use the dropdown item

File: libs/environment/src/components/network-switcher/network-switcher.tsx (hunk 2 above)
Comment (notbot00) [reply libs/environment/src/components/network-switcher/network-switcher.tsx]: this got removed due to merging master

File: libs/environment/src/components/network-switcher/network-switcher.tsx (hunk 3 above)
Comment (notbot00) [reply libs/environment/src/components/network-switcher/network-switcher.tsx]: sure, refactored it a little so i can use if blocks

File: apps/trading/components/navbar/navbar.tsx (hunk 1 above)
Comment (notbot00) [reply apps/trading/components/navbar/navbar.tsx]: absolutely, changed this to 24

Comment (mattrussell36) [reply apps/trading/components/navbar/navbar.tsx]: I think Sam meant to use `p-24` which is set up in our spacing definitions in the theme

File: apps/trading/components/navbar/navbar.tsx [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
-      <Link href="/" passHref={true}>
-        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
-        <a className="px-[26px]">
-          <Vega className="fill-white" />
-        </a>
-      </Link>
+      <div className="flex items-center h-full">
+        <Link href="/" passHref={true}>
+          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+          <a className="px-[24px]">
```
Comment (notbot00) [inline apps/trading/components/navbar/navbar.tsx:17]:
```suggestion
          <a className="px-24">
```

File: apps/trading/components/navbar/navbar.tsx (hunk 1 above)
Comment (notbot00) [reply apps/trading/components/navbar/navbar.tsx]: oh, i see, i didn't get that part. applied another fix.

**Pull request merged**